envoy/stream_info/stream_info.h

```diff
@@ -205,6 +205,16 @@ struct ResponseCodeDetailValues {
 
 using ResponseCodeDetails = ConstSingleton<ResponseCodeDetailValues>;
 
+struct StreamStateStringValues {
+  const std::string StreamStarted = "Started";
+  const std::string StreamInProgress = "InProgress";
+  const std::string StreamEnded = "Ended";
+};
+
+using StreamStateStrings = ConstSingleton<StreamStateStringValues>;
+
+enum StreamState { Started = 0x1, InProgress = 0x2, Ended = 0x4 };
```

Comment:
Can you elaborate on the precise meaning of Started/InProgress here - specifically that it is protocol-specific?

Reply:
Done